cmd/mattermost/commands/config.go

```diff
@@ -217,37 +211,26 @@ func configSetCmdF(command *cobra.Command, args []string) error {
 	configSetting := args[0]
 	newVal := args[1:]
 
-	// Update the config
-
-	// first disable the watchers
-	app.DisableConfigWatch()
-
 	// create the function to update config
 	oldConfig := app.Config()
 	newConfig := app.Config()
 	f := updateConfigValue(configSetting, newVal, oldConfig, newConfig)
 
-	// update the config
 	app.UpdateConfig(f)
-
-	// Verify new config
 	if err := newConfig.IsValid(); err != nil {
 		return err
 	}
 
-	if err := config.ValidateLocales(app.Config()); err != nil {
+	// UpdateConfig above would have already fixed these invalid locales, but we check again
```

Comment:
Invalid locale handling is... weird. I've tried to preserve the semantics for now.